clients/client/src/main/java/org/projectnessie/client/http/NessieHttpClient.java

```diff
@@ -93,43 +91,41 @@ private static void addTracing(HttpClient.Builder httpClient) {
     // It's safe to reference `GlobalTracer` here even without the required dependencies available
     // at runtime, as long as tracing is not enabled. I.e. as long as tracing is not enabled, this
     // method will not be called and the JVM won't try to load + initialize `GlobalTracer`.
-    Tracer tracer = GlobalTracer.get();
+    OpenTelemetry otel = GlobalOpenTelemetry.get();
+    Tracer tracer = otel.getTracer("Nessie");
     if (tracer != null) {
       httpClient.addRequestFilter(
           context -> {
-            Span span = tracer.activeSpan();
-            if (span != null) {
-              Span inner = tracer.buildSpan("Nessie-HTTP").start();
-              Scope scope = tracer.activateSpan(inner);
-              context.addResponseCallback(
-                  (responseContext, exception) -> {
-                    if (responseContext != null) {
-                      try {
-                        inner.setTag(
-                            "http.status_code", responseContext.getResponseCode().getCode());
-                      } catch (IOException e) {
-                        // There's not much we can (and probably should) do here.
-                      }
+            Span span =
+                tracer
+                    .spanBuilder("Nessie-HTTP")
+                    .startSpan()
+                    .setAttribute(SemanticAttributes.HTTP_URL, context.getUri().toString())
+                    .setAttribute(SemanticAttributes.HTTP_METHOD, context.getMethod().name())
+                    .setAttribute("component", "http");
+
+            @SuppressWarnings({"resource", "MustBeClosedChecker"})
+            Scope scope = span.makeCurrent();
```

Comment:
This is probably not necessary since we propagate headers manually (line 111). It might be simpler to avoid making the span current and inject `Context.current().with(span)` on line 111.

The "current" span concept is only about thread local stuff. It is not critical here since we do not expect any nested traceable events on this java thread during the request.